examples/eval.rs

```diff
@@ -0,0 +1,60 @@
+#![allow(non_upper_case_globals, non_camel_case_types, non_snake_case, improper_ctypes)]
+
+//! # Running scripts
+//! Here is the code under "Running scripts" in the MDN User Guide[1] translated into Rust. This
+//! only shows the ``run()`` function's contents because the original does as well.
+//!
+//! The actual code that is run is designed to be testable, unlike the example given.
+//! [1]: https://developer.mozilla.org/en-US/docs/Mozilla/Projects/SpiderMonkey/JSAPI_User_Guide
+//!
+
+
+use std::ptr;
+
+#[macro_use]
+extern crate mozjs;
+use mozjs::jsapi::*;
+use mozjs::jsval::UndefinedValue;
+use mozjs::rust::{JSEngine, RealmOptions, Runtime};
+use mozjs::rust::SIMPLE_GLOBAL_CLASS;
+
+fn run(rt: Runtime) {
+    unsafe {
+        let options = RealmOptions::default();
+        rooted!(in(rt.cx()) let global =
```

Comment:
We can probably do
```rust
rooted!(in(rt.cx()) let global = unsafe {
    JS_NewGlobalObject(...)
});
```
so the scope of the unsafety is reduced.

Reply:
I thought I tried that, but apparently not. Thanks!